go-controller/cmd/ovnkube/ovnkube.go

```diff
@@ -261,10 +297,16 @@ func runOvnKube(ctx *cli.Context) error {
 		metrics.StartMetricsServer(config.Kubernetes.MetricsBindAddress, config.Kubernetes.MetricsEnablePprof)
 	}
 
-	// run until cancelled
-	<-ctx.Context.Done()
-	close(stopChan)
-	factory.Shutdown()
+	// run until cancelled or error received
+	select {
+	case <-ctx.Context.Done():
+		close(stopChan)
+		factory.Shutdown()
+	case err := <-errChan:
+		klog.Error(err)
+		close(stopChan)
+		factory.Shutdown()
+	}
 	return nil
```

Comment:
@dave-tucker just do close(stopChan) + factory.Shutdown() after the select rather than twice inside?